src/py/flwr/superexec/exec_servicer.py

```diff
@@ -133,7 +134,24 @@ def StopRun(
     ) -> StopRunResponse:
         """Stop a given run ID."""
         log(INFO, "ExecServicer.StopRun")
-        raise NotImplementedError()
+        state = self.linkstate_factory.state()
+
+        # Exit if `run_id` not found
+        if not state.get_run(request.run_id):
+            context.abort(grpc.StatusCode.NOT_FOUND, "Run ID not found")
+
+        run_status = state.get_run_status({request.run_id})[request.run_id]
+
+        if run_status.status == Status.FINISHED:
+            log(ERROR, "Run ID `%s` is already finished", request.run_id)
+            return StopRunResponse(success=False)
```

Comment:
`context.abort` with an appropriate status code might be better here

Reply:
Fixed.